src/assets/scripts/client/game/ScoreController.js

```diff
@@ -0,0 +1,253 @@
+        EventBus.off(AIRCRAFT_EVENT.FULLSTOP, this._onLandingHandler);
+        EventBus.off(AIRCRAFT_EVENT.AIRSPACE_EXIT, this._onExitAirspaceHandler);
+
+        return this;
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onTakeoff
+     * @param aircraftModel {AircraftModel}
+     * @param runwayModel {RunwayModel}
+     */
+    _onTakeoff(aircraftModel, runwayModel) {
+        this._scoreWind(aircraftModel, 'taking off');
+        this._scoreRunwaySeparation(aircraftModel, runwayModel, 'taking off');
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onApproach
+     * @param aircraftModel {AircraftModel}
+     */
+    _onApproach(aircraftModel) {
+        this._penalizeLocalizerInterceptAltitude(aircraftModel);
+
+        // TODO: How can we evaluate the intercept angle?
+        // if () {
+        //     this._penalizeLocalizerInterceptAngle();
+        // }
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onLanding
+     * @param aircraftModel {AircraftModel}
+     * @param runwayModel {RunwayModel}
+     */
+    _onLanding(aircraftModel, runwayModel) {
+        this._scoreWind(aircraftModel, 'landed');
+        this._scoreRunwaySeparation(aircraftModel, runwayModel, 'landed');
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onExitAirspace
+     * @param aircraftModel {AircraftModel}
+     */
+    _onExitAirspace(aircraftModel) {
+        if (aircraftModel.isArrival()) {
+            this._onAirspaceExitForArrival(aircraftModel);
+
+            return;
+        }
+
+        if (aircraftModel.mcp.headingMode !== MCP_MODE.HEADING.LNAV) {
+            this._onAirspaceExitWithoutClearance(aircraftModel);
+
+            return;
+        }
+
+        this._onAirspaceExitWithClearance(aircraftModel);
+    }
+
+    /**
+     * An arriving aircraft is exiting the airpsace
+     *
+     * @for ScoreController
+     * @method _onAirspaceExitForArrival
+     * @param aircraftModel {AircraftModel}
+     * @private
+     */
+    _onAirspaceExitForArrival(aircraftModel) {
+        aircraftModel.radioCall('leaving radar coverage as arrival', AIRPORT_CONTROL_POSITION_NAME.APPROACH, true);
+        GameController.events_recordNew(GAME_EVENTS.AIRSPACE_BUST);
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onAirspaceExitWithClearance
+     * @param aircraftModel {AircraftModel}
+     * @private
+     */
+    _onAirspaceExitWithClearance(aircraftModel) {
+        aircraftModel.radioCall('switching to center, good day', AIRPORT_CONTROL_POSITION_NAME.DEPARTURE);
+        GameController.events_recordNew(GAME_EVENTS.DEPARTURE);
+    }
+
+    /**
+     * @for ScoreController
+     * @method _onAirspaceExitWithoutClearance
+     * @param aircraftModel {AircraftModel}
+     * @private
+     */
+    _onAirspaceExitWithoutClearance(aircraftModel) {
+        aircraftModel.radioCall('leaving radar coverage without proper clearance', AIRPORT_CONTROL_POSITION_NAME.DEPARTURE, true);
```

Comment:
Let's take this opportunity to clarify this response (since people always ask about what it means), and change it to `leaving airspace without being on our route`

Reply:
done